### PR TITLE
DOC: Group optional dependencies by category

### LINKED ISSUE #39594: DOC: Group optional dependencies by relevant topic
#### Location of the documentation

URL: https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#optional-dependencies
File in GitHub: https://github.com/pandas-dev/pandas/blob/master/doc/source/getting_started/install.rst

#### Documentation problem

The list of optional dependencies is currently just listed in alphabetical order. However, I think it would be much more logical to group packages by topic (PyTables **reads/writes hdf5** files, xlrd **reads Excel** files). Then you can compare/contrast which packages you should install for *your* usecase, e.g. should I install openpyxl or xlrd to read my excel files; I should research more about them.

Additionally, it might be helpful to include more detailed information about how they're different, but I am not as familiar with the minute differences between them. That will have to be for another time.

#### Suggested fix for documentation

I am willing to work on this.

I could either a) add a fourth column to the dependency table listing which group it is a part of (read/write HTML, read/write Excel), or b) split it into multiple tables with a minor header for each group.

Are there any issues with making this change, and which of those two actions should I prefer?

### PR DESCRIPTION
**Pull request by nmay231**
- [x] closes #39594 
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This PR splits the Optional Dependencies table into multiple tables one for each category. It should help people compare different options when looking for a specific capability, e.g. "I need to read an Excel file. What are my options?".

The specific grouping felt pretty subjective. If you think of a better way to group them, we can do that instead.

### CODE REVIEW

**Comment by jreback** [comment]
looks fine. can you post a rendered screenshot?

**Comment by nmay231** [comment]
@jreback 

Here's a [pastebin](https://pastebin.com/EJtracLm) of the full html page. And here's a couple screenshots.

File: doc/source/getting_started/install.rst [hunk 1]
```diff
@@ -255,47 +255,44 @@ For example, :func:`pandas.read_hdf` requires the ``pytables`` package, while
...
-fastparquet               0.4.0              Parquet reading / writing
-gcsfs                     0.6.0              Google Cloud Storage access
-html5lib                  1.0.1              HTML parser for read_html (see :ref:`note <optional_html>`)
-lxml                      4.3.0              HTML parser for read_html (see :ref:`note <optional_html>`)
-matplotlib                2.2.3              Visualization
+Jinja2                    2.10               Conditional formatting with DataFrame.style
+tabulate                  0.8.7              Printing in Markdown-friendly format (see `tabulate`_)
 numba                     0.46.0             Alternative execution engine for rolling operations
+                                             (see :ref:`Enhancing Performance <enhancingperf.numba>`)
+xarray                    0.12.3             pandas-like API for N-dimensional data
```
Comment (jorisvandenbossche) [inline doc/source/getting_started/install.rst:270]: I would maybe put jinja, tabulate and xarray in a different table, as those are not used in compute anything in pandas (or make plots), but for conversion to / represenation as another format (formatted string, styled html, or xarray object)

File: doc/source/getting_started/install.rst [hunk 2]
```diff
@@ -320,3 +317,44 @@ top-level :func:`~pandas.read_html` function:
...
+
+SQL databases and other data sources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+========================= ================== =============================================================
+Dependency                Minimum Version    Notes
+========================= ================== =============================================================
+SQLAlchemy                1.3.0              SQL support for databases other than sqlite
+psycopg2                  2.7                PostgreSQL engine for sqlalchemy
+pymysql                   0.8.1              MySQL engine for sqlalchemy
```
Comment (jorisvandenbossche) [inline doc/source/getting_started/install.rst:329]: The SQL dependencies are maybe worth having separate, and then a "Other data sources" or so?

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for working on this! 
I added two small suggestions

File: doc/source/getting_started/install.rst (hunk 1 above)
Comment (nmay231) [reply doc/source/getting_started/install.rst:270]: Yeah... I was basically trying to create a miscellaneous section without directly calling it misc. :sweat_smile: 

I'll just buckle down and label it Miscellaneous unless you have a different suggestion? I just didn't want to split it into two/three items per table.

**Comment by jreback** [comment]
@nmay231 thanks, yeah looks good. ping after @jorisvandenbossche suggestions are pushed & green.

**Comment by nmay231** [comment]
@jorisvandenbossche

**Comment by jreback** [comment]
can you merge master

**Comment by nmay231** [comment]
@jreback Merged with master

File: doc/source/getting_started/install.rst [hunk 3]
```diff
@@ -255,47 +255,44 @@ For example, :func:`pandas.read_hdf` requires the ``pytables`` package, while
 optional dependency is not installed, pandas will raise an ``ImportError`` when
 the method requiring that dependency is called.
 
+Miscellaneous Enhancments
```
Comment (jreback) [inline doc/source/getting_started/install.rst:258]: can you call this visualization & computation

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by nmay231** [comment]
@jreback Updated

File: doc/source/getting_started/install.rst [hunk 4]
```diff
@@ -255,47 +255,44 @@ For example, :func:`pandas.read_hdf` requires the ``pytables`` package, while
...
 SciPy                     1.12.0             Miscellaneous statistical functions
-xlsxwriter                1.0.2              Excel writing
-blosc                     1.17.0             Compression for HDF5
-fsspec                    0.7.4              Handling files aside from local and HTTP
-fastparquet               0.4.0              Parquet reading / writing
-gcsfs                     0.6.0              Google Cloud Storage access
-html5lib                  1.0.1              HTML parser for read_html (see :ref:`note <optional_html>`)
-lxml                      4.3.0              HTML parser for read_html (see :ref:`note <optional_html>`)
-matplotlib                2.2.3              Visualization
+Jinja2                    2.10               Conditional formatting with DataFrame.style
```
Comment (jreback) [inline doc/source/getting_started/install.rst:266]: can you split this table into visualization (matplotlib, jinja2, tabulate) and a separate computation (xarray, scipy, numba)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/getting_started/install.rst (hunk 4 above)
Comment (nmay231) [reply doc/source/getting_started/install.rst:265]: My bad. :facepalm: Did it just now.

**Comment by nmay231** [comment]
@jreback Sorry for the misunderstanding. It's fixed now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @nmay231 very nice.

https://pandas.pydata.org/pandas-docs/dev/ will have the built docs (once CI is done), check that everything renders properly.

thanks!

**Pull request merged**